Add checking to display the lowest price according to taxons

GSadee opened the pull request.

Rafikooo commented on src/Application/Checker/ProductVariantLowestPriceDisplayChecker.php:
```diff
@@ -0,0 +1,49 @@
+
+final class ProductVariantLowestPriceDisplayChecker implements ProductVariantLowestPriceDisplayCheckerInterface
+{
+    public function displayLowestPrice(ProductVariantInterface $productVariant, array $context): bool
```
This method name is a bit misleading. At first glance, IMO the name fits the void function rather than bool
```suggestion
    public function shouldDisplayLowestPrice(ProductVariantInterface $productVariant, array $context): bool
```
```suggestion
    public function isLowestPriceDisplayable(ProductVariantInterface $productVariant, array $context): bool
```
Or similar, wdyt?

Rafikooo commented:
I am not sure if this is the right case, but when we can add MAIN_TAXON as excluded then descendant taxons do not inherit this configuration. In the following example after adding "/api/v2/admin/taxons/dresses" it works fine
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/40125720/223674407-dac7c564-d07b-42c4-ab42-d7babc0b573f.png">
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/40125720/223674565-53ff067b-6da0-42f8-af44-83c7998c2567.png">

Rafikooo commented:
Since I'm not sure what should do with MAIN_TAXON (whether we should even allow such a taxon to be added to the excluded), here's I provided a better example:
All T-Shirts excluded
<img width="1427" alt="image" src="https://user-images.githubusercontent.com/40125720/223677918-d61b2960-fd95-45d4-b1a9-dac777cec715.png">
But the lowest price is present:
<img width="1433" alt="image" src="https://user-images.githubusercontent.com/40125720/223678495-8604b707-d24b-4b1f-89bc-9631980ab6f1.png">

GSadee commented:
> Since I'm not sure what should do with MAIN_TAXON (whether we should even allow such a taxon to be added to the excluded), here's I provided a better example: All T-Shirts excluded <img alt="image" width="1427" src="https://user-images.githubusercontent.com/40125720/223677918-d61b2960-fd95-45d4-b1a9-dac777cec715.png"> But the lowest price is present: <img alt="image" width="1433" src="https://user-images.githubusercontent.com/40125720/223678495-8604b707-d24b-4b1f-89bc-9631980ab6f1.png">

You are right, I implemented this functionality as simple as possible and overlooked the case of checking descendants for excluded taxons. I added the relevant scenarios and adjusted the implementation in PHP. However, ideally, this would probably be refactored to use DQL for such a query, but I would prefer to do that later in a separate PR.

And about the main taxon, IMO, it should not be taken into account here. It is actually mainly used for the menu in the catalog.

TheMilek commented on features/product/viewing_products/not_seeing_lowest_price_for_product_that_has_all_taxons_excluded_on_channel.feature:
```diff
@@ -0,0 +1,50 @@
+        And this product's price changed to "$10.00" and original price changed to "$21.00"
+        And the store also has a product "T-Shirt Watermelon" priced at "$22.00"
+        And it belongs to "T-Shirts"
+        And the store also has a product "T-Shirt PHP" priced at "$23.00"
+        And it belongs to "T-Shirts"
+        And it belongs to "PHP"
```
```suggestion
        And the store also has a product "T-Shirt PHP" priced at "$23.00" that belongs to "T-Shirts" and "PHP"
```
OR maybe?
```suggestion
        And the store also has a product "T-Shirt PHP" priced at "$23.00"
        And it belongs to "T-Shirts" and "PHP"
```

GSadee replied:
Yup, you are right, I agree that this would be better, but when I could I based it on the steps that are defined in Sylius/Sylius and I did not want to add a whole new context just for this one step

Rafikooo commented:
Thank you, @GSadee!
